json_journal/data_analysis.py

```diff
@@ -82,3 +83,49 @@ def total_profit(full_account_name_balances):
             pass
     total = (income * (-1)) - expenses
     return total
+
+
+def cumulative_profit(ledger):
+    '''
+    Calculate the cumulative profit.
+
+    Parameters
+    ----------
+    Ledger : dict
+        Loaded json file.
+
+    Returns
+    -------
+    Dates : list
+        List of dates.
+    Cumulative_profit : list
+        List of cumulative total of profit.
+    '''
+    sorted_ledger = sorted(ledger, key=lambda d: d['date'])
+    dates = []
+    profits = []
+    for transaction in sorted_ledger:
+        transaction_date = transaction['date']
+        entries = transaction['entries']
+        profit = 0
+        for entry in entries:
+            amount = entry['amount']
+            account = entry['account']
+            if (account.startswith("expense:") or account.startswith("income")):
+                profit -= amount
+            else:
+                pass
+
+        if transaction_date in dates:
```

Comment:
```suggestion
        if transaction_date == dates[-1]:
```

Reply:
Let's take advantage of that sorted array

Reply:
done, but needed to add an extra if statement to cover the first case when the list is empty

Reply:
I tried changing to make it for the next transaction date but couldn't figure it out :(

Reply:
done